### ISSUE TITLE
Flag use of var when never assigned to.

### ISSUE THREAD

**Issue by MarkCLewis** [scalastyle/scalastyle#312]
I'd like to see a rule that flags the use of a var that could have been a val. So any var declaration that is never assigned to later would be flagged. It would be fine if it only worked for local variables and private members so modification would be constrained to the same file. I realize this is a more complex rule to implement because the extended scope of usage might make it impractical, but I figured I would ask.

**Comment by matthewfarwell** [comment]
You are right that this is complicated, but it might actually be possible for a limited scope. The trouble is, you run into scoping rules very rapidly.

For example, if you declare a var in a class, then declare a val of the same name in a method in the class, then this would be a false negative, if I'm not mistaken. Maybe.

**Comment by MarkCLewis** [comment]
I don't think the val declaration should register as an assignment since it would be detectable as a declaration, but you would get a false negative if the local were a var and it were assigned to and the scope of the local var wasn't differentiated from the class level var. Because of shadowing, the outer var doesn't even have to be class level. One local var could shadow another local var and if either is assigned to, it might be challenging to detect that the other isn't.
